cfme/automate/explorer/instance.py

```diff
@@ -336,6 +249,89 @@ def delete_if_exists(self):
             self.delete()
 
 
+@attr.s
+class InstanceCollection(BaseCollection):
+
+    ENTITY = Instance
+
+    @property
+    def tree_path(self):
+        return self.parent.tree_path
+
+    def create(self, name=None, display_name=None, description=None, fields=None, cancel=False):
+        add_page = navigate_to(self, 'Add')
+        fill_dict = {
+            k: v
+            for k, v in {
+                'name': name,
+                'display_name': display_name,
+                'description': description,
+                'fields': fields,
+            }.items()
+            if v is not None}
+        add_page.fill(fill_dict)
+        if cancel:
+            add_page.cancel_button.click()
+            add_page.flash.assert_no_error()
+            add_page.flash.assert_message('Add of new Automate Instance was cancelled by the user')
+            return None
+        else:
+            add_page.add_button.click()
+            add_page.flash.assert_no_error()
+            add_page.flash.assert_message('Automate Instance "{}" was added'.format(name))
+            return self.instantiate(
+                name=name,
+                display_name=display_name,
+                description=description,
+                fields=fields)
+
+    def delete(self, *instances):
+        all_page = navigate_to(self.parent, 'Details')
+        all_page.instances.select()
+        instances = list(instances)
+        parents = set()
+        for instance in instances:
+            parents.add(instance.parent)
+        if len(parents) > 1:
+            raise ValueError('You passed instances that are not under one class.')
+
+        checked_instances = []
+        if not all_page.instances.table.is_displayed:
+            raise ValueError('No instance found!')
+        all_page.instances.table.uncheck_all()
+        for row in all_page.instances.table:
```

Comment:
This doesn't account for multiple pages, I think we're going to need a generic method to handle this so we don't end up repeating this loop in multiple collection delete methods.